### DISCUSSION TITLE
Why is it that when declaring a file field from a form model, the content-type in the swagger is defined as application/x-www-form-urlencoded?

### DISCUSSION THREAD

**Discussion by D1maD1m0nd** [fastapi/fastapi#12510] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class GenerationImageFormData(BaseModel):
    Id: str = Form(
        title="Unique identifier for generating the image that was obtained as a result of the request "
              "get_generation_types from the list",
    ),
    Image: UploadFile = File(
        title="Source image downloaded by user"),
    TgUserId: str = Form(title="Telegram user id")



@router_generation.post(
    "/generate_image/",
    response_model_exclude_none=True,
    description="Generate image in fitting room",

)
async def generate_image(
       data : Annotated[GenerationImageFormData, Form()],
) -> dict:

    return {}
```

### Description

Hello, I converted all requests with forms into form models, each of the models has a field with files. When I go to the Swagger UI and make a request, I see in the headers content-type equal to **application/x-www-form-urlencoded** and not **multipart/form-data**, so the request returns error validation.
![image](https://github.com/user-attachments/assets/82521008-7e3c-49aa-b89b-52259992f12a)

Is there a workaround so that the content type is determined correctly?

### Operating System

Windows

### Operating System Details

windows 11

### FastAPI Version

0.115.3

### Pydantic Version

2.9.2

### Python Version

3.12

**Comment by Krismix1** [comment, marked as answer]
I've stumbled upon the same problem when trying the new "form model" feature with files.
Given that it works when not using form models, I suspect this to be a bug (the feature is fairly new).

For the time being, I have found a workaround for this - set the `media_type` argument:
```py
async def generate_image(
       data : Annotated[GenerationImageFormData, Form(media_type="multipart/form-data")],
) -> dict:
```
